### PR TITLE
Add Overwrite property to UnzipArchive

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Currently if you try to extract a file over an existing file it failes with an IOException.

### CODE REVIEW

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs [hunk 1]
```diff
@@ -54,6 +54,30 @@ public void UnzipsFile()
...
+
+            var task = new UnzipArchive
+            {
+                File = dest,
+                Destination = outDir,
+                BuildEngine = new MockEngine(),
+                Overwrite = true
+            };
+
+            Assert.True(task.Execute(), "Task failed");
```
Comment (JunTaoLuo) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: Failed?

Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: This is the message that appears when the assertion fails. A better message might be "The task failed but should have passed".

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs [hunk 2]
```diff
@@ -54,6 +54,30 @@ public void UnzipsFile()
...
+        [Fact]
+        public void Overwrites()
+        {
+            var files = new[]
+            {
+                "a.txt",
+                "dir/b.txt"
+            };
+
+            var dest =CreateZip(files);
```
Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: Nit: formatting

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs [hunk 3]
```diff
@@ -54,6 +54,30 @@ public void UnzipsFile()
...
+            var task = new UnzipArchive
+            {
+                File = dest,
+                Destination = outDir,
+                BuildEngine = new MockEngine(),
+                Overwrite = true
+            };
+
+            Assert.True(task.Execute(), "Task failed");
+            Assert.True(task.Execute(), "Task failed on second run.");
```
Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: Would be good to assert it actually wrote to the file. Before you run the task, create `$outDir/a.txt` with contents. Assert it is overwritten as an empty file after the task runs.

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs (hunk 1 above)
Comment (JunTaoLuo) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: My bad, when I see two values in an `Assert` call, I automatically assume it's `Assert.Equals`. Yea this makes sense now.

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs [hunk 4]
```diff
@@ -54,6 +54,30 @@ public void UnzipsFile()
             }
         }
 
+        [Fact]
+        public void Overwrites()
```
Comment (JunTaoLuo) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: Add a test that make sure no overwrite occurs when it's set to false?

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs (hunk 1 above)
Comment (ryanbrandenburg) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: I just copied the message from the other tests here. I'll update them too.

**Comment by ryanbrandenburg** [comment]
This isn't needed for rel/2.0.0 work, it's part of work to clean up some of our more obtuse MSBuild files.

**Comment by ryanbrandenburg** [comment]
:up::date:

File: test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs [hunk 5]
```diff
@@ -54,6 +54,68 @@ public void UnzipsFile()
...
+                File = dest,
+                Destination = outDir,
+                BuildEngine = new MockEngine(),
+                Overwrite = true
+            };
+
+            Directory.CreateDirectory(outDir);
+
+            // Create a.txt before trying to unzip
+            using (var file = File.CreateText(Path.Combine(outDir, "a.txt")))
```
Comment (natemcmaster) [inline test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs]: This still isn't exactly verifying the file is overwritten. Try this:
```c#
var path = Path.Combine(outDir, "a.txt");
File.WriteAllText("contents!", path);
Assert.True(task.Execute(), "The task failed but should have passed.");
Assert.Empty(File.ReadAllText(path));
```

**Comment by ryanbrandenburg** [comment]
:up::date:

**Pull request merged**